src/smoothscroll.js

```diff
@@ -89,11 +89,23 @@
      * @returns {Node} el
      */
     function findScrollableParent(el) {
+      var isBody;
+      var hasScrollableSpace;
+      var hasVisibleOverflow;
+
       do {
         el = el.parentNode;
-      } while (el !== d.body
-              && !(el.clientHeight < el.scrollHeight
-              || el.clientWidth < el.scrollWidth));
+
+        // set condition variables
+        isBody = el === d.body;
```

Comment:
We can bail if the closest parent of the element is the body. That said I don't know in how many cases this will actually happen so I'm not seeing a real benefit but if you think is worth it I'll add at top @iamdustan